s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -58,6 +60,32 @@ private[alpakka] final class S3Stream(credentials: AWSCredentials, region: Strin
   val MinChunkSize = 5242880 //in bytes
   val signingKey = SigningKey(credentials, CredentialScope(LocalDate.now(), region, "s3"))
 
+  /** listBucket lists files in a bucket, filtered by an optional prefix 
+    * 
+    * @param bucket name of the bucket
+    * @param prefix optional prefix
+    * @return A akka Source with keys
+    * 
+    */
+  def listBucket(bucket: String, prefix: Option[String] = None): Source[String, NotUsed] = {
```

Comment:
Good deal. With that in mind, I think I am more of a fan of your implementation. There's some stuff in mine about the URLs that I still need to land. 

One last thing. S3 is known to fail for for random reasons and most of the time the prescribed way to fix it is to retry. With that, would it make sense to put a recoverWithRetries somewhere in this call chain to deal with failed calls? Or should that be something that happens on using the source?